core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -3713,6 +3714,152 @@ protected PaginatedSearchResult doListUsers(String filter, int limit, int offset
+        } catch (UserStoreException ex) {
+            handleGetUserCountFailure(UserCoreErrorConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_GETTING_COUNT_USERS
+                            .getCode(),
+                    String.format(UserCoreErrorConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_GETTING_COUNT_USERS
+                                    .getMessage(),
+                            ex.getMessage()), claimUri, value);
+            throw ex;
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            String ErrorMsg = "Error while getting attribute name from " + claimUri ;
+            throw new UserStoreException(ErrorMsg, e);
+        }
+    }
+
+    /**
+     * Count roles in user stores.
+     *
+     * @param filter the filter for the user name. Use '*' to have all.
+     * @return user count
+     * @throws UserStoreException UserStoreException
+     */
+    public long doCountRoles(String filter) throws UserStoreException {
+
+        long usersCount = 0;
+        String sqlStmt;
+        if (filter.startsWith(UserCoreConstants.INTERNAL_DOMAIN)) {
+            sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.COUNT_INTERNAL_ROLES);
+            String names[] = filter.split(UserCoreConstants.DOMAIN_SEPARATOR);
+            filter = names[1].trim();
+        } else if (filter.startsWith(UserCoreConstants.APPLICATION_DOMAIN)) {
+            sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.COUNT_APPLICATION_ROLES);
+        } else {
+            sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.COUNT_ROLES);
+        }
+
+        if (StringUtils.isNotEmpty(filter)) {
+            filter = filter.trim();
+            filter = filter.replace("*", "%");
+            filter = filter.replace("?", "_");
+        } else {
+            filter = "%";
+        }
+
+        try (Connection dbConnection = getDBConnection();
+             PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
+
+            prepStmt.setString(1, filter);
+            if (sqlStmt.toUpperCase().contains(UserCoreConstants.SQL_ESCAPE_KEYWORD)) {
+                prepStmt.setString(2, SQL_FILTER_CHAR_ESCAPE);
+                if (sqlStmt.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                    prepStmt.setInt(3, tenantId);
+                }
+            } else {
+                if (sqlStmt.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                    prepStmt.setInt(2, tenantId);
+                }
+            }
+
+            ResultSet resultSets = prepStmt.executeQuery();
+                while (resultSets.next()) {
+                    return resultSets.getLong(1);
+                }
+
+        } catch (SQLException e) {
+            String msg = "Error occurred while retrieving users for filter : " + filter;
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new UserStoreException(msg, e);
```

Comment:
use error codes

Reply:
fix - in https://github.com/wso2/carbon-kernel/pull/2375

Reply:
fixed in  - https://github.com/wso2/carbon-kernel/pull/2480